### PR TITLE
[For Review] CAP-234/fix/consumer-home-v2 :: add controllers

### PR DESCRIPTION
**Pull request by laubana**

### CODE REVIEW

File: controllers/discountController.js [hunk 1]
```diff
@@ -114,13 +114,15 @@ const getActiveDiscounts = async (req, res) => {
         });
 
         if (!activeDiscounts?.length) {
-            return res.status(400).json({ error: "No active coupon" });
+            return res
+                .status(400)
+                .json({ error: "Active discounts not found" });
         }
```
Comment (dacosio) [inline controllers/discountController.js:120]: @laubana this should not give you an error when discounts are empty. It just means that there is no record of active discounts yet.

File: controllers/menuDiscountController.js [hunk 2]
```diff
@@ -39,23 +44,54 @@ const getAllMenuDiscount = async (req, res) => {
...
             .populate({
                 path: "discountId",
-                select: "label percentDiscount"
+                select: "label percentDiscount",
             })
             .lean();
 
         if (!menuDiscounts?.length) {
             return res.status(400).json({ message: "No menu discounts found" });
         }
```
Comment (dacosio) [inline controllers/menuDiscountController.js:54]: @laubana same case here, an empty collection is not an error. It's just that it is empty. So, return 200, and then the empty menuDiscounts. Remove lines 52 to 54.

File: controllers/menuDiscountController.js [hunk 3]
```diff
@@ -39,23 +44,54 @@ const getAllMenuDiscount = async (req, res) => {
...
+                path: "discount",
+            })
+            .populate({
+                path: "merchant",
+            })
+            .lean();
+
+        if (!menuDiscounts?.length) {
+            return res.status(400).json({ error: "Menu discounts not found" });
+        }
```
Comment (dacosio) [inline controllers/menuDiscountController.js:84]: same case here with the comments above.

File: controllers/ratingController.js [hunk 4]
```diff
@@ -24,13 +23,11 @@ const getAllMerchantRatings = async (req, res) => {
             .lean();
 
         if (!ratings?.length) {
-            return res.status(400).json({ error: "No rating" });
+            return res.status(400).json({ error: "Ratings not found" });
         }
```
Comment (dacosio) [inline controllers/ratingController.js:27]: same feedback as the aformentioned.

File: controllers/usersController.js [hunk 5]
```diff
@@ -23,7 +23,15 @@ const getAllConsumers = async (req, res) => {
...
-        const merchants = await Merchant.find()
+        const { keyword } = req.query;
+
+        console.log(keyword);
+
+        const merchants = await Merchant.find(
+            keyword && {
+                name: { $regex: keyword, $options: "i" },
+            }
+        )
```
Comment (dacosio) [inline controllers/usersController.js:34]: is this for search?

File: routes/menuDiscountRoutes.js [hunk 6]
```diff
@@ -7,5 +7,9 @@ router
     .route("/menuDiscount")
     .post(menuDiscountController.addMenuDiscount)
```
Comment (dacosio) [inline routes/menuDiscountRoutes.js:8]: transform menuDiscount to kebab-case

**Review by dacosio — CHANGES_REQUESTED** [review]

File: routes/menuDiscountRoutes.js (hunk 6 above)
Comment (laubana) [reply routes/menuDiscountRoutes.js:8]: I didn't change this.

File: controllers/menuDiscountController.js (hunk 2 above)
Comment (laubana) [reply controllers/menuDiscountController.js:54]: I didn't change this.

File: .eslintcache [hunk 7]
```diff
@@ -1 +1 @@
...
...12","messages":"13","suppressedMessages":"14","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"22i0oo",{"filePath":"15","messages":"16","suppressedMessages":"17","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"18","messages":"19","suppressedMessages":"20","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"21","messages":"22","suppressedMessages":"23","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"24","messages":"25","suppressedMessages":"26","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/yh/Documents/GitHub/capstone_backend/models/Merchant.js",[],[],"/Users/yh/Documents/GitHub/capstone_backend/controllers/discountController.js",[],[],"/Users/yh/Documents/GitHub/capstone_backend/controllers/menuDiscountController.js",[],[],"/Users/yh/Documents/GitHub/capstone_backend/controllers/ratingController.js",[],[],"/Users/yh/Documents/GitHub/capstone_backend/controllers/usersController.js",[],[]]
```
Comment (dacosio) [inline .eslintcache:1]: do not commit eslintcache

File: controllers/menuDiscountController.js [hunk 8]
```diff
@@ -39,23 +44,56 @@ const getAllMenuDiscount = async (req, res) => {
...
             .populate({
                 path: "discountId",
-                select: "label percentDiscount"
+                select: "label percentDiscount",
             })
             .lean();
 
         if (!menuDiscounts?.length) {
             return res.status(400).json({ message: "No menu discounts found" });
         }
```
Comment (dacosio) [inline controllers/menuDiscountController.js:53]: must be status 200 with {menuDiscounts}

**Review by dacosio — CHANGES_REQUESTED** [review]

**Review by dacosio — CHANGES_REQUESTED** [review]
Remove duplicate logics as discussed.

**Review by dacosio — APPROVED** [review]

**Pull request merged**